### DISCUSSION TITLE
Dimmer not switch OFF when "targetValue/set: 'OFF'"

### DISCUSSION THREAD

**Discussion by hestiap** [zwave-js/zwave-js-ui#2360] [Q&A]
zjs2m reveived "targetValue/set: 'OFF'" from with mqtt from a Qubino  Mini Dimmer  ZMNHHD
but the switch stayed ON

The log in zjs2m
```
2022-04-09 13:01:46.102 INFO ZWAVE: Node 150: value updated: 38-0-currentValue 2 => 2
2022-04-09 13:01:46.579 INFO MQTT: Message received on zwave/150/38/0/targetValue/set: 'OFF'
2022-04-09 13:01:46.581 INFO ZWAVE: Writing 0 to 150-38-0-targetValue
2022-04-09 13:01:46.656 INFO ZWAVE: Node 150: value updated: 38-0-targetValue 0 => 0
...
2022-04-09 13:01:48.925 INFO ZWAVE: Node 150: value updated: 50-0-value-66049 0 => 0
2022-04-09 13:01:50.709 INFO ZWAVE: Node 150: value updated: 38-0-currentValue 2 => 2
```
![image](https://user-images.githubusercontent.com/73014953/162569209-a641e8ac-5b23-470d-a3f7-17cf13996143.png)

Is this normal?

**Comment by robertsLando** [comment]
@hestiap Could you send driver logs please? Giving that there is no error on my side I think there is something on driver

cc @AlCalzone

**Comment by AlCalzone** [comment]
a) It looks like the device didn't fully turn off, or it sent an incorrect update:
```
2022-04-09 13:01:50.709 INFO ZWAVE: Node 150: value updated: 38-0-currentValue 2 => 2
```

b) Binary switch and multilevel switch aren't linked in any way in the driver, since they can belong to different unrelated features of the device. In general it is pretty strange to have both at the same time.

This looks a lot like this issue: https://github.com/zwave-js/node-zwave-js/issues/4247
The device can be switched from Binary (On/Off) to Multilevel (Dimmer) working mode (requiring an exclusion-inclusion cycle) and it seems that wasn't completely done here.

**Comment by graphicsxp** [comment]
just following because i've encountered the exact [same issue](https://github.com/zwave-js/zwavejs2mqtt/issues/2362) yesterday. 
Only difference is that i don't send to same endpoint. I use `zwave/light-dressing/38/1/Off/set`  instead of targetValue.
In any case, this reduces the currentValue to the minValue (which cannot be set to 0) and which in my case is 7. Instead it should just turn off the light.

In my home assistant dashboard, I added the Fibaro as a light. If I toggle it on/off the device it works fine, but I guess it does not go through mqtt in this case.

also it does not look similar to 4247, at least in my case toggling the wall switch updates the fibaro status just fine.

The only problem I found is to turn it off through mqtt.

**Comment by hestiap** [comment]
@robertsLando 
cc @AlCalzone 

> Could you send driver logs please

I wasn't able to reproduce the issue.
When it happened, it took me some time to switch off the light, I think I had to dim the light to a big value before to a small value, and try several times (the dimming was on 2% as the [150-112-0-60] Minimum dimming value was on  something like 15%)
The issue was the same with the GUI and the switch in the wall.
I think I could not investigate more to give you information.
If I find it again I'll try to set the driver logs

**Comment by graphicsxp** [comment]
Ok, I think I will turn on/off like the @hestiap is doing, that is sending the value 'Off' or 'On' to the targetValue queue. 
There is a bug with the Off or On queues as mentioned above, but it doesn't bother me if there is a workaround.
